controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/Maintainer.java

```diff
@@ -27,16 +30,23 @@
     private final JobControl jobControl;
     private final ScheduledExecutorService service;
     private final String name;
+    private List<SystemName> systemNames;
```

Comment:
I would make this a `Set<SystemName` with a default value of `EnumSet.allOf(SystemName.class)`. That way you can simplify the check below to just `systemNames.contains(controller.system())`.